tests/integration_tests/test_conversion.py

```diff
@@ -0,0 +1,659 @@
+# coding=utf-8
+from __future__ import print_function, absolute_import, unicode_literals
+
+import json
+import tempfile
+import unittest
+
+import os
+from unittest import TestCase
+import requests
+import shutil
+import time
+
+from general_tools import file_utils
+from manager.manager import TxManager
+from general_tools.file_utils import unzip
+from aws_tools.s3_handler import S3Handler
+from client.client_webhook import ClientWebhook
+from bs4 import BeautifulSoup
+
+COMMIT_LENGTH = 40
+USE_WEB_HOOK_LAMBDA = True
+
+class TestConversions(TestCase):
+    """
+    To test locally, you must set two environment variables:
+
+        set TEST_DEPLOYED to "test_deployed"
+        set GOGS_USER_TOKEN to tx-manager user token
+
+    Integration test will run on dev unless TRAVIS_BRANCH is set to 'master' (then will run on prod)
+    """
+
+    def setUp(self):
+        branch = os.environ.get("TRAVIS_BRANCH", "develop") # default is testing develop branch (dev)
+
+        destination = "dev-" # default
+        if branch == "master":
+            destination = "" # no prefix for production
+
+        self.destination = destination
+        self.api_url = 'https://{0}api.door43.org'.format(destination)
+        self.pre_convert_bucket = '{0}tx-webhook-client'.format(destination)
+        self.gogs_url = 'https://git.door43.org'.format(destination)
+        self.cdn_bucket = '{0}cdn.door43.org'.format(destination)
+        self.job_table_name = '{0}tx-job'.format(destination)
+        self.module_table_name = '{0}tx-module'.format(destination)
+        self.cdn_url = 'https://{0}cdn.door43.org'.format(destination)
+
+        print("Testing on '" + branch + "' branch, e.g.: " + self.api_url)
+
+    def tearDown(self):
+        """Runs after each test."""
+        # delete temp files
+        if hasattr(self, 'temp_dir') and os.path.isdir(self.temp_dir):
+            shutil.rmtree(self.temp_dir, ignore_errors=True)
+
+    @unittest.skip("Skipping broken conversion that needs to be fixed - preconvert leaves backslash at end of line")
```

Comment:
@richmahn:

Unfortunately it still fails to generate output. Not a lot helpful in the log:

errors": {"L": [{"S": "RequestId: f49fe014-45fd-11e7-b7c1-a9e0bc04871c Process exited before completing request"}]}, "log": {"L": [{"S": "Started job 2e9d433f47fb8942846d2a6b934319394582a1dc331e673dc10d04bab3d7b794 at 2017-05-31T12:37:48Z"}, {"S": "Telling module usfm2html to convert https://s3-us-west-2.amazonaws.com/dev-tx-webhook-client/preconvert/db971b4985.zip and put at https://dev-cdn.door43.org/tx/job/2e9d433f47fb8942846d2a6b934319394582a1dc331e673dc10d04bab3d7b794.zip"}, {"S": "Conversion failed"}, {"S": "Finished job 2e9d433f47fb8942846d2a6b934319394582a1dc331e673dc10d04bab3d7b794 at 2017-05-31T12:37:50Z"}]}, "source": {"S": "https://s3-us-west-2.amazonaws.com/dev-tx-webhook-client/preconvert/db971b4985.zip"}, "status": {"S": "failed"}